src/main/java/au/com/codeka/carrot/bindings/JsonArrayBindings.java

```diff
@@ -0,0 +1,67 @@
+package au.com.codeka.carrot.bindings;
+
+import au.com.codeka.carrot.Bindings;
+import au.com.codeka.carrot.ValueHelper;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+
+/**
+ * {@link Bindings} based on the values in a {@link JSONArray}.
+ *
+ * @author Marten Gajda
+ */
+public final class JsonArrayBindings implements Bindings, Iterable
```

Comment:
I'm thinking, given that we now have an easily-extensible interface that maybe this sort of specific sub-class could be kept out of the main library. Maybe we have a separate jar that implements this object for JSON data objects? I'm just thinking I'd like to keep the dependencies of the main library to a minimum (for example, if you're using Gson instead of org.json or something, then this would just be extra deps for no reason).

What do you think?